aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -404,9 +441,53 @@ void ImagingResampleHorizontalConvolution8u4x(
       -1, 15, -1, 11, -1, 14, -1, 10, -1, 13, -1, 9, -1, 12, -1, 8
     )
   };
+  const __m256i masks_low_high_c3[2] = {
+    _mm256_set_epi8(
+      -1, -1, -1, -1, -1, 5, -1, 2, -1, 4, -1, 1, -1, 3, -1, 0,
+      -1, -1, -1, -1, -1, 5, -1, 2, -1, 4, -1, 1, -1, 3, -1, 0
+    ),
+    _mm256_set_epi8(
+      -1, -1, -1, -1, -1, 11, -1, 8, -1, 10, -1, 7, -1, 9, -1, 6,
+      -1, -1, -1, -1, -1, 11, -1, 8, -1, 10, -1, 7, -1, 9, -1, 6
+    )
+  };
+
+  const auto mask_low = (num_channels == 3) ? masks_low_high_c3[0] : masks_low_high_c4[0];
+  const auto mask_high = (num_channels == 3) ? masks_low_high_c3[1] : masks_low_high_c4[1];
+
+  const auto stride = num_channels * 1;  // num channels * sizeof(uint8)
+
+  TORCH_INTERNAL_ASSERT(stride == 3 || stride == 4);
+
+  // Precompute xmax limits for block 4 and block 2
+  // lineIn0 + stride * (x + xmin) + 16 <= lineIn0 + stride * (xmax + xmin)
+  // --> x <= xmax - 16.0 / stride
```

Comment:
Where does 16 come from here?

Reply:
In the "block 4" we are reading data with `_mm_loadu_si128`, i.e. 16 bytes` per one line.
I'll improve the comments here.